### PR TITLE
Replaces .env file with env vars in README

### LINKED ISSUE #120: Stop using the odd `.env.tpl` file
As explained here:
https://github.com/src-d/code-annotation/pull/36#discussion_r167946866

The usage of env file, populated with empty env vars is odd, and not very Devops friendly since handly configuration files is a something to be avoided, even if it contains env vars.

The env vars should be explicitly defined in all the examples, and be omitted if contains the default value or are empty, in the same that this should be explicitly defined in the `.travis.yml.

### PR DESCRIPTION
**Pull request by carlosms**
Fixes #120

All the env vars are explained in a table. The .env file is still suggested for development in CONTRIBUTING.

The TODOs in the table need to be with the new vars from https://github.com/src-d/code-annotation/pull/145.

### CODE REVIEW

File: README.md
Comment (smacker) [inline README.md]: When I `view` this file the table is rendered like this:
screen shot 2018-02-26 at 18 28 41

File: README.md
Comment (smacker) [inline README.md]: Here should be example how to set required env variables. At least:
- CAT_OAUTH_CLIENT_ID
- CAT_OAUTH_CLIENT_SECRET

are there any other?

Comment (smacker) [reply README.md]: maybe we also need to provide information how to run it with DB.
Currently, it will run the server with empty DB which isn't very usable.

File: README.md
Comment (smacker) [inline README.md:97]: is default variable ok? If not it should be in `docker run` too.

File: README.md
Comment (carlosms) [reply README.md]: Fixed

File: CONTRIBUTING.md
Comment (smacker) [inline CONTRIBUTING.md]: Sorry, for me personally this sentence isn't clear.

Maybe we should rephrase like:
> For convenience, you can configure `.env` file with all the environment variables described in the [README](./README.md).
> We provide a template with the configuration suitable for development in `.env.tpl`. You can copy it to `.env` and set up GitHub OAuth credentials there.

The text isn't perfect, just to give you better understanding what I mean. What do you think?

File: README.md
Comment (smacker) [inline README.md]: todo? should we write a description here?

Comment (dpordomingo) [inline README.md]: `TODO` value could be
>`hostname:port` where the API is served

File: README.md
Comment (dpordomingo) [inline README.md]: `TODO` value could be
>`hostname:port` where the frontend is served

File: README.md
Comment (smacker) [inline README.md]: `Installation` and `Environment Variables` have the same size of header. It looks a bit weird:
screen shot 2018-02-26 at 18 44 48

File: README.md
Comment (dpordomingo) [inline README.md]: I'd add the link:
[syntax of the CAT_DB_CONNECTION](#set-the-internal-database-connection)

File: README.md
Comment (smacker) [inline README.md:27]: people often move not mandatory variables somewhere in the bottom of README. Currently, the table is larger than my display and we are going to have more... Just as an idea for improvement.

File: README.md
Comment (carlosms) [reply README.md]: Done. Regarding your second comment, I think the reordering of the steps falls into #67, to be done in a future PR.

File: README.md
Comment (dpordomingo) [inline README.md]: I'd add a link to `#github-oauth-tokens`

File: README.md
Comment (dpordomingo) [inline README.md]: I'd add a link to `#github-oauth-tokens`

File: README.md
Comment (dpordomingo) [inline README.md]: I'd add a link to `#access-control`

File: README.md
Comment (dpordomingo) [inline README.md]: I'd add a link to `#access-control`

File: README.md
Comment (carlosms) [reply README.md:97]: I guess the default is OK... otherwise it should not be the default value.

**Review by dpordomingo — APPROVED** [review]
LGTM, but could you add some links?

File: README.md
Comment (dpordomingo) [reply README.md]: Imo, teaching to the user how to provide a db is out of scope for this PR, and should be done by #67

File: README.md
Comment (carlosms) [reply README.md]: Yes, let's add the description now.
Can you please tell me the meaning of SERVER_URL, STATIC_PROXY_TARGET after the changes made in #145 ?

File: README.md
Comment (smacker) [reply README.md:97]: I believe it's not. Opening `/var/code-annotation/internal.db` will fail with error: directory doesn't exist.

We have 3 options:
- change defaults
- change documentation
- add `mkdir` in Dockerfile (I prefer this one)

File: README.md
Comment (dpordomingo) [reply README.md:27]: Since this `README.md` is already huge and imho it does not strictly follow our [recommendations for `README.md`](https://github.com/src-d/guide/blob/master/engineering/documentation.md#readmemd) I would not care too much about the length of the table: I'd merge as soon as possible, and improve `README.md` when facing issue  https://github.com/src-d/code-annotation/issues/67

I'd reorder the mandatory ones in a way it appears first, but it's up to you @carlosms

File: README.md
Comment (dpordomingo) [inline README.md]: I'd add a link to `#access-control`

File: README.md
Comment (dpordomingo) [inline README.md]: I'd add a link to `#access-control`

File: README.md
Comment (smacker) [reply README.md]: `SERVER_URL ` doesn't change the meaning. It's server url available from outside. Frontend is using it to make http calls.

`STATIC_PROXY_TARGET` not sure we should document it. Normally nobody should change it.
And if they do it should be somehow be reflected in create-react-app. It should set HOST & PORT variables. https://github.com/facebook/create-react-app/blob/next/packages/react-scripts/scripts/start.js#L56

#145 isn't merged yet, it's one of the issues we need to resolve there. Maybe we will remove this variable and replace with something else.

File: README.md
Comment (dpordomingo) [inline README.md]: I'd add a link to `#github-oauth-tokens`

File: README.md
Comment (dpordomingo) [inline README.md]: I'd add a link to `#github-oauth-tokens`

**Review by dpordomingo — APPROVED** [review]
LGTM, but I'd add some links

File: README.md
Comment (carlosms) [reply README.md:27]: Thanks both for the feedback, I moved the mandatory vars to the top of the table.

File: CONTRIBUTING.md
Comment (bzz) [inline CONTRIBUTING.md]: `For convenience, you can create a ..`

File: CONTRIBUTING.md
Comment (bzz) [inline CONTRIBUTING.md]: What do you think about changing
`you will need a .env file ... to run the app in dev mode` -> `you will need to customize the configuration, either by setting env vars, or though an .env file .... to `

to communicate more clearly the optional nature of .env file?

File: README.md
Comment (bzz) [inline README.md]: It's always good to explain a bit of rationale for a need, so what do you think about re-phrasing by adding:
`In order for authentication \w Github to work on your environment, you need to set up OAuth application on Github platform`.

**Review by dpordomingo — APPROVED** [review]
LGTM, imo you can merge once you've addressed @bzz suggestions

File: README.md
Comment (smacker) [reply README.md:97]: @carlosms please address it somehow before merging (maybe just mention in another issue)

Comment (carlosms) [reply README.md:97]: I created issue https://github.com/src-d/code-annotation/issues/173 so we don't loose track of this.

**Review by dpordomingo — APPROVED** [review]
Many thanks Carlos

**Comment by dpordomingo** [comment]
Now that Carlos already addressed all the provided feedback, could you PTAL @bzz @smacker ?

**Review by bzz — APPROVED** [review]
Awesome 📜 !

File: README.md
Comment (smacker) [inline README.md]: please remove it. #145 may never be merged in master or this variable will be replaced by something

**Comment by dpordomingo** [comment]
@carlosms Let's remove `CAT_STATIC_PROXY_TARGET` as suggested by [prev comment](https://github.com/src-d/code-annotation/pull/170#discussion_r171217541), squash, rebase and merge :)

**Pull request merged**
